src/ui/uploader.js

```diff
@@ -28,13 +28,20 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
         members: {
             currentFile: null
         },
-        // Holds a fileObjectURL for file preview purposes
         model: {
-            fileObjectURL: null,
-            fileDetails: null
+            // uploadState can be one of the following values:
+            // "ready" (the initial state), "uploading", "errorReceived"
+            uploadState: "ready",
+            fileObjectUrl: null, // for the file preview
+            previousFileObjectUrl: null, // for file deletion on change/upload
```

Comment:
We can probably remove the need for the `previosFileObjectUrl` model path. Instead we can add a model listener for changes to the fileObjectURL and trigger a delete when it changes. Or there is a request to re-upload actually fire a delete immediately for the current fileObjectUrl if one exists. We would need to update the middleware that handles the upload. This will also make things more RESTful.

Reply:
Would it be reasonable to count this as part of the work for [SJRK-416](https://issues.fluidproject.org/browse/SJRK-416)? If so, I can adjust the description of that Jira to explicitly include the file upload endpoints.

Reply:
@BlueSlug is that because of the DELETE operation?

Reply:
That's right: since making this change will require changing the server endpoints, it might make sense to do it at that same time